ArtOfIllusion/src/artofillusion/procedural/ProceduralTextureCommon.java

```diff
@@ -0,0 +1,151 @@
+
+package artofillusion.procedural;
+
+import artofillusion.MaterialPreviewer;
+import artofillusion.TextureParameter;
+import artofillusion.image.ImageMap;
+import artofillusion.math.RGBColor;
+import artofillusion.texture.Texture;
+import static artofillusion.texture.Texture.*;
+import artofillusion.ui.ActionProcessor;
+import artofillusion.ui.Translate;
+import artofillusion.ui.ValueSelector;
+import buoy.event.ValueChangedEvent;
+import buoy.widget.BDialog;
+import buoy.widget.BorderContainer;
+import buoy.widget.LayoutInfo;
+import buoy.widget.RowContainer;
+import java.awt.Rectangle;
+
+/**
+ *
+ * @author MaksK
+ */
+public class ProceduralTextureCommon
+{
+  private ProceduralTextureCommon()
+  {    
+  }
+  
+  /**
+   * Create a Procedure object for texture.
+   */
+
+  public static Procedure createTextureProcedure()
+  {
+    return new Procedure(new OutputModule [] {
+      new OutputModule(Translate.text("Diffuse"), Translate.text("white"), 0.0, new RGBColor(1.0f, 1.0f, 1.0f), IOPort.COLOR),
+      new OutputModule(Translate.text("Specular"), Translate.text("white"), 0.0, new RGBColor(1.0f, 1.0f, 1.0f), IOPort.COLOR),
+      new OutputModule(Translate.text("Transparent"), Translate.text("white"), 0.0, new RGBColor(1.0f, 1.0f, 1.0f), IOPort.COLOR),
+      new OutputModule(Translate.text("Emissive"), Translate.text("black"), 0.0, new RGBColor(0.0f, 0.0f, 0.0f), IOPort.COLOR),
+      new OutputModule(Translate.text("Transparency"), "0", 0.0, null, IOPort.NUMBER),
+      new OutputModule(Translate.text("Specularity"), "0", 0.0, null, IOPort.NUMBER),
+      new OutputModule(Translate.text("Shininess"), "0", 0.0, null, IOPort.NUMBER),
+      new OutputModule(Translate.text("Roughness"), "0", 0.0, null, IOPort.NUMBER),
+      new OutputModule(Translate.text("Cloudiness"), "0", 0.0, null, IOPort.NUMBER),
+      new OutputModule(Translate.text("BumpHeight"), "0", 0.0, null, IOPort.NUMBER),
+      new OutputModule(Translate.text("Displacement"), "0", 0.0, null, IOPort.NUMBER)});
+  }
+  
+  public static TextureParameter[] getTextureParameters(Object texture, Procedure proc)
+  {
+    Module modules[] = proc.getModules();
+    int count = 0;
+
+    for (Module module1 : modules)
+      if (module1 instanceof ParameterModule)
+        count++;
+    
+    TextureParameter params[] = new TextureParameter [count];
+    count = 0;
+    for (Module module : modules)
+      if (module instanceof ParameterModule)
+      {
+        params[count] = ((ParameterModule) module).getParameter(texture);
+        ((ParameterModule) module).setIndex(count++);
+      }
+    return params;
+    
+  }
+  
+  /** Determine whether given procedure texture has a non-zero value anywhere for a particular component.
+   *  @param proc the Procedure to check 
+      @param component    the texture component to check for (one of the Texture *_COMPONENT constants)
+  */  
+  public static boolean hasTextureComponent(Procedure proc, int component)
```

Comment:
Same here. Putting these in a separate class adds some friction to reading and _understanding_ the Procedural Texture code. Not sure if the reduced line count is enough to warrant that.